lintcheck/src/main.rs

```diff
@@ -336,36 +333,67 @@ impl Crate {
 
         let shared_target_dir = clippy_project_root().join("target/lintcheck/shared_target_dir");
 
-        let mut args = if config.fix {
+        let mut cargo_clippy_args = if config.fix {
             vec!["--fix", "--"]
         } else {
             vec!["--", "--message-format=json", "--"]
         };
 
+        let mut clippy_args = Vec::<&str>::new();
         if let Some(options) = &self.options {
             for opt in options {
-                args.push(opt);
+                clippy_args.push(opt);
             }
         } else {
-            args.extend(&["-Wclippy::pedantic", "-Wclippy::cargo"])
+            clippy_args.extend(&["-Wclippy::pedantic", "-Wclippy::cargo"])
         }
 
         if lint_filter.is_empty() {
-            args.push("--cap-lints=warn");
+            clippy_args.push("--cap-lints=warn");
         } else {
-            args.push("--cap-lints=allow");
-            args.extend(lint_filter.iter().map(|filter| filter.as_str()))
+            clippy_args.push("--cap-lints=allow");
+            clippy_args.extend(lint_filter.iter().map(|filter| filter.as_str()))
+        }
+
+        if let Some(server) = server {
+            let target = shared_target_dir.join("recursive");
```

Comment:
this does set a different target dir for --recursive, right? did you run into some problems when not doing this?

Reply:
It wouldn't cause a problem, IIRC I was trying to avoid the regular `lintcheck`'s target dir getting removed unnecessarily. But as it may get deleted anyway on account of `clippy`'s timestamp maybe it's not really making a difference doing this